Signal/src/ViewControllers/ConversationView/ConversationViewController.m

```diff
@@ -3001,31 +3001,23 @@ - (void)sendMessageAttachment:(SignalAttachment *)attachment
     }
 }
 
-- (void)sendContactShare:(ContactShareViewModel *)contactShare
+- (void)sendContactShare:(OWSContactShare *)contactShare
```

Comment:
We now convert proposed contact shares to actual contact shares (e.g. write a `TSAttachment`) at the last moment before sending.

We do this:

* ...to avoid cluttering the database with incomplete shares.
* ...to simplify the "cleanup contact share attachments on disk" logic.
* ...to avoid touching the db or creating transactions until we need to.

I've added a separate `sendProposedContactShare` method to make this more clear.